### PR TITLE
Add golangci-lint workflow

### PR DESCRIPTION
**Pull request by SuperQ**
## 💸 TL;DR
Add GitHub action workflow for running golangci-lint.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: install Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: 1.22.x
```
Comment (davinci26) [inline .github/workflows/golangci-lint.yml:29]: We use

```
module github.com/reddit/monoceros

go 1.19
```

should we match and update after?

Comment (SuperQ) [reply .github/workflows/golangci-lint.yml:29]: The `go` directive in Go modules is a minimum version specifier for compiler compatibility.

We could setup a matrix job to test all versions, but I've not found that necessary for running golangci-lint.

For `go test`, we'll want to setup a compatibility test matrix.

Comment (davinci26) [reply .github/workflows/golangci-lint.yml:29]: TIL, makes sense

**Review by davinci26 — APPROVED** [review]
LGTM

**Pull request merged**